### PR TITLE
Add a Duration answer type

### PR DESCRIPTION
**Pull request by kierenbeckett**
### What is the context of this PR?
We want to be able to collect durations in years/months.  See also:

ONSDigital/go-launch-a-survey#101
ONSDigital/eq-schema-validator#62

### How to review 
Run test_durations.json, run through the survey and check the output in `/dump/submission`.  The current behaviour for complex types in v1 payloads is to `str()` them.

### CODE REVIEW

File: scripts/test_schemas.sh [hunk 1]
```diff
@@ -6,8 +6,8 @@ if [ "$1" == "--local" ] || [ "$2" == "--local" ]; then
 fi
 
 if [ "$run_docker" == true ]; then
-    docker pull onsdigital/eq-schema-validator
-    validator="$(docker run -d -p 5001:5000 onsdigital/eq-schema-validator)"
+    docker pull onsdigital/eq-schema-validator:add_year_month_duration
```
Comment (kierenbeckett) [inline scripts/test_schemas.sh]: These need reverting once schema PR is merged

**Review by dcdarrell9 — APPROVED** [review]
LGTM 👍

File: app/translations/messages.pot [hunk 2]
```diff
@@ -8,14 +8,28 @@ msgid ""
 msgstr ""
 "Project-Id-Version: PROJECT VERSION\n"
 "Report-Msgid-Bugs-To: EMAIL@ADDRESS\n"
-"POT-Creation-Date: 2018-07-11 14:50+0100\n"
+"POT-Creation-Date: 2018-07-13 14:26+0100\n"
```
Comment (dwyeradam) [inline app/translations/messages.pot]: Well remembered.

File: scripts/test_schemas.sh (hunk 1 above)
Comment (dwyeradam) [reply scripts/test_schemas.sh]: This can be reverted now.

File: tests/app/forms/test_date_form.py [hunk 3]
```diff
@@ -52,6 +53,85 @@ def test_generate_year_date_form_creates_empty_form(self):
...
+
+        self.assertIsNone(form().data)
+
+    def test_date_form_format_data(self):
+        schema = load_schema_from_params('test', 'dates')
+        error_messages = schema.error_messages
+
+        answers = schema.get_answers_by_id_for_block('date-block')
+
+        data = {'field': '2000-01-01'}
```
Comment (dwyeradam) [inline tests/app/forms/test_date_form.py:94]: Just a little thing (my little OCD regarding test dates) but can the DD field be greater than 12. Shows and tests clearly that the date format is being correctly used throughout.

**Review by dwyeradam — APPROVED** [review]
Nice refactoring here as well. Can we ensure the commit description includes comments around refactor as well as just addition of duration type please.

**Comment by dwyeradam** [comment]
@kierenbeckett should be a simple conflict to fix and then we can merge this.

**Pull request merged**